util/ranger/detacher.go

```diff
@@ -48,7 +48,7 @@ func detachColumnCNFConditions(conditions []expression.Expression, checker *cond
 		accessConditions = append(accessConditions, cond)
 		if checker.shouldReserve {
 			filterConditions = append(filterConditions, cond)
-			checker.shouldReserve = false
+			checker.shouldReserve = checker.length != types.UnspecifiedLength
```

Comment:
Why change this?

Reply:
`detachColumnXNFConditions` will be reused when calculate index access conditions.